examples/metrics_simple.cpp

```diff
@@ -23,18 +23,18 @@ int main()
     abacus::metric_info infos[4] = {
         abacus::metric_info{name0, "Fuel consumption in kilometers per liter",
                             abacus::metric_type::float64,
-                            abacus::metric_kind::constant},
+                            abacus::metric_kind::constant, "km/l"},
         abacus::metric_info{name1, "Wheels on the car",
                             abacus::metric_type::uint64,
-                            abacus::metric_kind::constant},
+                            abacus::metric_kind::constant, "wheels"},
         abacus::metric_info{name2,
                             "Days until next maintenance, if less than 0, "
                             "maintenance is overdue",
                             abacus::metric_type::int64,
-                            abacus::metric_kind::gauge},
+                            abacus::metric_kind::gauge, "days"},
         abacus::metric_info{name3, "Is the car registered",
                             abacus::metric_type::boolean,
-                            abacus::metric_kind::gauge}};
+                            abacus::metric_kind::gauge, ""}};
```

Comment:
maybe we could add "" as the default parameter.. and it would actually read nicer if we made it strongly typed...

so:

        abacus::metric_info{name2,
                            "Days until next maintenance, if less than 0, "
                            "maintenance is overdue",
                            abacus::metric_type::int64,
                            abacus::metric_kind::gauge, abacus::unit{"days"}},
                            
                            
What do you think?

Looking at the code above it would be even greater with `abacus::type` and `abacus::kind`, so dropping ``metric_``.